server/src/main.py

```diff
@@ -179,3 +168,53 @@ async def delete_file(workspace_id: str, path: str):
     # -- for a later date --
     # TODO: Check user has write permission for workspace
     # see @stat()
+
+
+@app.post("/workspaces/{workspace_id}/directory/{directory_path}")
+async def create_directory(workspace_id: str, directory_path: str):
+    empty_data = bytes()
+    data_stream = io.BytesIO(empty_data)
+    client.put_object(
+        workspace_id,
+        directory_path + "/",
+        data=data_stream,
+        length=0,
+        content_type="application/octet-stream",
+    )
+    return {"message": f"CREATED {directory_path} in {workspace_id}"}
+
+
+@app.delete(
+    "/workspaces/{workspace_id}/directory/{directory_path}",
+    status_code=status.HTTP_204_NO_CONTENT,
+)
+async def delete_directory(workspace_id: str, directory_path: str):
+    errors_count: int = 0
+    del_list: list[str] = []
+
+    objects = list((client.list_objects(workspace_id, prefix=directory_path + "/")))
+
+    for i in objects:
+        logger.debug(f"Adding {i.object_name} to delete list")
+        del_list.append(DeleteObject(i.object_name))
+    errors = client.remove_objects(
+        workspace_id,
+        del_list,
```

Comment:
We could replace this with `[DeleteObject(object.object_name) for object in objects]` and replace instances of `len(del_list)` with `len(objects)`.